src/ng-select/lib/ng-dropdown-panel.component.ts

```diff
@@ -196,11 +196,14 @@ export class NgDropdownPanelComponent implements OnInit, OnChanges, OnDestroy {
         // it's important first to check if it has vertical scrollbar since this._renderer.addClass(this._dropdown, BOTTOM_CSS_CLASS)
         // could make it to appear
         let hasVerticalScrollbar = false;
+        let documentHasVerticalScrollbar = false;
 
         if (this.appendTo) {
           hasVerticalScrollbar = this._parent.scrollHeight > this._parent.clientHeight;
         }
 
+        documentHasVerticalScrollbar = document.documentElement.scrollHeight !== document.documentElement.clientHeight;
```

Comment:
Can you explain why you added this? Is there some extra UI issues without this fix?

Reply:
Yup
So in the case element you are appending to doesn't have a scrollbar, but globally there is a scrollbar on the entire document. In that case, hasVerticalScrollbar would return false (because the parent doesn't have it), but on the other side, the entire document has it so we need to cover that case as well.